### ISSUE TITLE
`next/cache`'s `unstable_cache` can not be used, `generateStaticParams` is not executed in AsyncLocalStorage context

### ISSUE THREAD

**Issue by SukkaW** [vercel/next.js#50765]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 22.4.0: Mon Mar  6 21:00:17 PST 2023; root:xnu-8796.101.5~3/RELEASE_X86_64
    Binaries:
      Node: 18.13.0
      npm: 8.19.3
      Yarn: 1.22.19
      pnpm: 8.5.1
    Relevant packages:
      next: 13.4.5-canary.4
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
      typescript: N/A
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true)

### Link to the code that reproduces this issue or a replay of the bug

https://github.com/SukkaW/next-unstable-cache-repro

### To Reproduce

```bash
git clone https://github.com/SukkaW/next-unstable-cache-repro && cd next-unstable-cache-repro && npm i && npm run dev
```

Then open http://localhost:3000/a

### Describe the Bug

```
- error Error: Invariant: incrementalCache missing in unstable_cache async ()=>{
    await new Promise((resolve)=>setTimeout(resolve, 5000));
    return {
        title: "Hello World"
    };
}
    at unstable_cache (webpack-internal:///(sc_server)/./node_modules/next/dist/server/web/spec-extension/unstable-cache.js:19:15)
    at eval (webpack-internal:///(sc_server)/./src/app/[slug]/data.js:9:75)
    at (sc_server)/./src/app/[slug]/data.js (/Users/sukka/Project/next-unsateble-cache-repro/.next/server/app/[slug]/page.js:2229:1)
    at __webpack_require__ (/Users/sukka/Project/next-unsateble-cache-repro/.next/server/webpack-runtime.js:33:42)
    at eval (webpack-internal:///(sc_server)/./src/app/[slug]/page.js:9:63)
    at (sc_server)/./src/app/[slug]/page.js (/Users/sukka/Project/next-unsateble-cache-repro/.next/server/app/[slug]/page.js:2240:1)
    at Function.__webpack_require__ (/Users/sukka/Project/next-unsateble-cache-repro/.next/server/webpack-runtime.js:33:42)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async collectGenerateParams (/Users/sukka/Project/next-unsateble-cache-repro/node_modules/next/dist/build/utils.js:822:17) {
  digest: undefined
}
null
```

### Expected Behavior

The page should render.

### How are you deploying your application? (if relevant)

next dev

**Comment by SukkaW** [comment]
cc @shuding

**Comment by SheryConcepts** [comment]
facing the same issue

**Comment by Djboy08** [comment]
same here, I am facing the same issue. Even in instances where I don't have any static page generation I find myself getting this error.

**Comment by SukkaW** [comment]
> Same issue here, when using unstable_cache in a **separate** file. Inlining the call inside of a page does work.

I attempted to move the invocation onto the page, but it still doesn't seem to be functioning properly. How did you solve the issue?

**Comment by ifeanyiisitor** [comment]
Facing this issue as well

**Comment by JulianJorgensen** [comment]
Same here

**Comment by joulev** [comment]
I don't know when this issue was fixed, but as of v13.4.20-canary.23 the issue has been fixed.

<img width="585" alt="image" src="https://github.com/vercel/next.js/assets/44609036/04150fef-9a27-4bc0-b897-749720a1c5f9">

**Comment by SukkaW** [comment]
@joulev Confirm `next-unstable-cache-repro` no longer throws. Closing the issue.
